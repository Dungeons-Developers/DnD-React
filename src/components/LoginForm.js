import React from 'react';
import { connect } from 'react-redux';

import TextField from '@material-ui/core/TextField';
import Button from '@material-ui/core/Button';
import Box from '@material-ui/core/Box';

import useForm from '../hooks/useForm'

import { login } from '../store/slices/user-slice';

function LoginForm({ login }) {
  const fields = {
    username: '',
    password: '',
  };

  const { handleChange, handleSubmit } = useForm(fields);

  function submit(e) {
    e.preventDefault();
    handleSubmit(login);
    e.target.reset();
  }

  function formChange(e) {
    handleChange(e.target.id, e.target.value);
  }

  return (
    <form className='login-form' noValidate autoComplete="off" onSubmit={submit}>
      <div>
        <TextField id="username" label="Username" fullWidth onChange={formChange} />
      </div>
      <div>
        <TextField id="password" label="Password" fullWidth onChange={formChange} type="password" />
      </div>
      <Box my="1rem">
        <Button variant="contained" color="primary" fullWidth type='submit'>
          Log In
        </Button>
      </Box>
    </form >
  );
}

const mapDispatchToProps = { login };

export default connect(null, mapDispatchToProps)(LoginForm);
