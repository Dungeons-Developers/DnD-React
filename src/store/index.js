import { configureStore } from '@reduxjs/toolkit';
import { combineReducers } from 'redux';

// import slices here

import users from './slices/user-slice';
import campaign from './slices/campaign-slice';
import characters from './slices/character-slice';
import theme from './slices/theme-slice';

let reducers = combineReducers( { users, campaign, characters, theme } );

let store = configureStore( { reducer: reducers } );

export default store;

