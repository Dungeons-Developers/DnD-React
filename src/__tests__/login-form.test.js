import React from 'react';
import {mount} from 'enzyme';

import {Provider} from 'react-redux';
import store from '../store';

import LoginForm from '../components/LoginForm';

describe('login form tests', () => {

  let component = mount(
    <Provider store={store}>
      <LoginForm />
    </Provider>
  );

  it('contains initial components', () => {
    expect(component.find('.login-form')).toBeDefined();
    expect(component.find('#username')).toBeDefined();
    expect(component.find('#password')).toBeDefined();
    expect(component.find('button')).toBeDefined();
  });

  it('matches snapshot?', () => {
    expect(component).toMatchSnapshot();
  });

});