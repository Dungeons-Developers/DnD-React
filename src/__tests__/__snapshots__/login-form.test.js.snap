// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`login form tests matches snapshot? 1`] = `ReactWrapper {}`;
